nixos/modules/services/continuous-integration/radicle/adapters/native.nix

```diff
@@ -0,0 +1,141 @@
+    instances = lib.mkOption {
+      type = lib.types.attrsOf (
+        lib.types.submodule (
+          { config, name, ... }:
+          {
+            options = {
+              enable = lib.mkEnableOption "this radicle-native-ci instance" // {
+                default = true;
+                example = false;
+              };
+
+              name = lib.mkOption {
+                type = lib.types.str;
+                description = ''
+                  Adapter name that is used in the radicle-ci-broker configuration.
+                  Defaults to the attribute name.
+                '';
+              };
+
+              package = lib.mkPackageOption pkgs "radicle-native-ci" { };
+
+              runtimePackages = lib.mkOption {
+                type = lib.types.listOf lib.types.package;
+                description = "Packages added to the adapter's {env}`PATH`.";
+                defaultText = lib.literalExpression ''
+                  with pkgs; [
+                    bash
+                    coreutils
+                    curl
+                    gawk
+                    gitMinimal
+                    gnused
+                    wget
+                  ]
+                '';
+              };
+
+              settings = lib.mkOption {
+                type = lib.types.submodule {
+                  freeformType = settingsFormat.type;
+
+                  options = {
+                    state = lib.mkOption {
+                      type = lib.types.path;
+                      description = "Directory where per-run directories are stored.";
+                      defaultText = lib.literalExpression ''"''${config.services.radicle.ci.broker.stateDir}/adapters/native/${config.name}"'';
+                    };
+
+                    log = lib.mkOption {
+                      type = lib.types.path;
+                      description = "File where radicle-native-ci should write the run log.";
+                      defaultText = lib.literalExpression ''"''${config.services.radicle.ci.broker.logDir}/adapters/native/${config.name}.log"'';
+                    };
+
+                    base_url = lib.mkOption {
+                      type = lib.types.nullOr lib.types.str;
+                      description = "Base URL for build logs (mandatory for access from CI broker page).";
+                      default = null;
+                    };
+                  };
+                };
+                description = ''
+                  Configuration of radicle-native-ci.
+                  See <https://app.radicle.xyz/nodes/seed.radicle.xyz/rad:z3qg5TKmN83afz2fj9z3fQjU8vaYE#configuration> for more information.
+                '';
+                default = { };
+              };
+            };
+
+            config = {
+              name = lib.mkDefault name;
+
+              runtimePackages = with pkgs; [
+                bash
+                coreutils
+                curl
+                gawk
+                gitMinimal
```

Comment:
Should we use config.programs.git.package?

Reply:
Hm, neither this module nor the radicle module (which also uses `gitMinimal`) depend on the git module, so I'm not sure it would make sense to use this option here. If you really want to use a different git package, you can just override the `runtimePackages` option in your config.

Reply:
I am more thinking about deduplicating gitMinimal and git but if you say it makes no sense, I trust you.